### PR TITLE
feat: add moneygram cash out

### PR DESCRIPTION
**Pull request by Merculiar**
Fixes https://github.com/espresso-cash/espresso-cash-public/issues/1511

## Changes

implements moneygram cash out

## Videos

<details>
<summary>Withdraw </summary>

https://github.com/espresso-cash/espresso-cash-public/assets/46864104/6495a0b7-dee9-4814-a4fd-be060476fd48

</details>

<details>
<summary>Refund</summary>

https://github.com/espresso-cash/espresso-cash-public/assets/46864104/6aecffaa-d4da-45b3-9c58-c51f134a264a

</details>

## Checklist

- [ ] PR is ready for review (if not, it should be a draft).
- [x] PR title follows [Conventional Commits][1] guidelines.
- [x] Screenshots/video added.
- [ ] Tests added.
- [x] Self-review done.

[1]: https://www.conventionalcommits.org/en/v1.0.0/

### CODE REVIEW

**Review by ookami-kb — APPROVED** [review]
@Merculiar I'm merging the PR now to release a new version, please resolve the issues in the separate PR.

File: packages/espressocash_app/lib/features/ramp/screens/off_ramp_order_screen.dart [hunk 1]
```diff
@@ -122,49 +140,99 @@ class OffRampOrderScreenContent extends StatelessWidget {
         ),
       OffRampOrderStatus.insufficientFunds =>
         '${context.l10n.splitKeyErrorMessage2} ${context.l10n.errorMessageInsufficientFunds}',
+      OffRampOrderStatus.preProcessing ||
+      OffRampOrderStatus.postProcessing =>
+        'Preparing your withdrawal',
+      OffRampOrderStatus.ready => 'Open Moneygram to continue',
```
[the remark this answers is not in the document]
Comment (Merculiar) [reply packages/espressocash_app/lib/features/ramp/screens/off_ramp_order_screen.dart:146]: Regardless of 'Ready' status, it was added to moneygram only. When we withdraw, we need to wait before usdc is already bridged from solana to stellar and only after  that can we proceed user to open the moneygram iframe. So when we confirm the money has been received, we proceed status to ready and open iframe button appears for user to continue.

[the remark this answers is not in the document]
Comment (Merculiar) [reply packages/espressocash_app/lib/features/ramp/screens/off_ramp_order_screen.dart:146]: Maybe it is better to rename it to something like "WithdrawalUrlReady" /  "readyToContinue"? Open to suggestions.

File: packages/espressocash_app/lib/features/ramp/services/off_ramp_order_service.dart [hunk 2]
```diff
@@ -290,12 +330,13 @@ class OffRampOrderService implements Disposable {
       RampPartner.kado => sl<KadoOffRampOrderWatcher>(),
       RampPartner.scalex => sl<ScalexOffRampOrderWatcher>(),
       RampPartner.coinflow => sl<CoinflowOffRampOrderWatcher>(),
+      RampPartner.moneygram => null,
       RampPartner.rampNetwork ||
       RampPartner.moneygram ||
```
[the remark this answers is not in the document]
Comment (Merculiar) [reply packages/espressocash_app/lib/features/ramp/services/off_ramp_order_service.dart:335]: Fixed here https://github.com/espresso-cash/espresso-cash-public/pull/1533.

File: packages/espressocash_app/lib/features/activities/widgets/off_ramp_tile.dart [hunk 3]
```diff
@@ -1,3 +1,5 @@
+// ignore_for_file: avoid-wildcard-cases-with-enums
```
[the remark this answers is not in the document]
Comment (Merculiar) [reply packages/espressocash_app/lib/features/activities/widgets/off_ramp_tile.dart:1]: Fixed here: https://github.com/espresso-cash/espresso-cash-public/pull/1533

File: packages/espressocash_app/lib/features/ramp/screens/off_ramp_order_screen.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (Merculiar) [reply packages/espressocash_app/lib/features/ramp/screens/off_ramp_order_screen.dart:146]: https://github.com/espresso-cash/espresso-cash-public/pull/1533

**Pull request merged**